astropy/stats/bayesian_blocks.py

```diff
@@ -0,0 +1,505 @@
+- Irregularly-spaced point measurements via the :class:`PointMeasures` class
+
+For more fine-tuned control over the fitness functions used, it is possible
+to define custom :class:`FitnessFunc` classes directly and use them with
+the :func:`bayesian_blocks` routine.
+
+One common application of the Bayesian Blocks algorithm is the determination
+of optimal adaptive-width histogram bins. This uses the same fitness function
+as for irregularly-spaced time series events. The easiest interface for
+creating Bayesian Blocks histograms is the :func:`astropy.stats.histogram`
+function.
+
+References
+----------
+.. [1] http://adsabs.harvard.edu/abs/2012arXiv1207.5578S
+.. [2] http://astroML.org/ http://github.com/astroML/astroML/
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as np
+import inspect
+
+# TODO: implement other fitness functions from appendix B of Scargle 2012
+
+__all__ = ['FitnessFunc', 'Events', 'RegularEvents', 'PointMeasures',
+           'bayesian_blocks']
+
+
+def bayesian_blocks(t, x=None, sigma=None,
+                    fitness='events', **kwargs):
+    """Compute optimal segmentation of data with Scargle's Bayesian Blocks
+
+    This is a flexible implementation of the Bayesian Blocks algorithm
+    described in Scargle 2012 [1]_.
+
+    Parameters
+    ----------
+    t : array_like
+        data times (one dimensional, length N)
+    x : array_like (optional)
+        data values
+    sigma : array_like or float (optional)
+        data errors
+    fitness : str or object
+        the fitness function to use for the model.
+        If a string, the following options are supported:
+
+        - 'events' : binned or unbinned event data
+            extra arguments are ``p0``, which gives the false alarm probability
+            to compute the prior, or ``gamma`` which gives the slope of the
+            prior on the number of bins.
+        - 'regular_events' : non-overlapping events measured at multiples
+            of a fundamental tick rate, ``dt``, which must be specified as an
+            additional argument.  The prior can be specified through ``gamma``,
+            which gives the slope of the prior on the number of bins.
+        - 'measures' : fitness for a measured sequence with Gaussian errors
+            The prior can be specified using ``gamma``, which gives the slope
+            of the prior on the number of bins.  If ``gamma`` is not specified,
+            then a simulation-derived prior will be used.
+
+        Alternatively, the fitness parameter can be an instance of
+        :class:`FitnessFunc` or a subclass thereof.
+
+    **kwargs :
+        any additional keyword arguments will be passed to the specified
+        :class:`FitnessFunc` derived class.
+
+    Returns
+    -------
+    edges : ndarray
+        array containing the (N+1) edges defining the N bins
+
+    Examples
+    --------
+    Event data:
+
```

Comment:
I left that out because I thought it would be distracting. The numbers won't mean all that much to the reader of the documentation, and the separate unit tests take care of input/output testing.